cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -69,13 +74,22 @@ void MetricsCollector::Initialize(std::string metric_namespace,
 
 void MetricsCollector::SubscribeAllTopics()
 {
-  for (auto it = topics_.begin(); it != topics_.end(); ++it) {
-    auto sub = node_->create_subscription<ros_monitoring_msgs::msg::MetricList>(
-            *it, kNodeSubQueueSize,
-            [this](ros_monitoring_msgs::msg::MetricList::UniquePtr metric_list_msg) -> void {
+  for (auto it = topics_.cbegin(); it != topics_.cend(); ++it) {
```

Comment:
Consider using a C++11 range-based loop (https://en.cppreference.com/w/cpp/language/range-for) for better readability.

Reply:
Done.